src/setIntersection.js

```diff
@@ -0,0 +1,21 @@
+import dataTypeCheck from '../src/dataTypeCheck'
+export default function setInstersection(array, intersect) {
+
+  if(!dataTypeCheck(array,'array')) {
+    return "Wrong Data Type"
+  }
+
+let result = []
+  for(let i=0;i<array.length;i++) {
+    let intersection = false
```

Comment:
again, no need to define a temp variable for ```intersection```

if the the ```if``` condition is true then just push to your array.  

If you want to only push the first instance that this occurs then I believe you can ```return``` inside the the if conditional inside your for loop and that breaks you out of the loop.  I know for certain that returning inside an if conditional always ends the function. I am 95% certain this also happens inside a for loop.